### PR TITLE
Stop passing in auto-discovered xunit traits

### LINKED ISSUE #4105: TargetOSTrait to be available in non-CoreFX repos
Arcade's MSBuild SDK provides everything required to use `PlatformSpecificAttribute` downstreams, except for the `<TargetOSTrait>`, which is defined in corefx specific targets.

For context: https://github.com/dotnet/core-setup/pull/8510#discussion_r333996226

Can this property group be moved partly from: https://github.com/dotnet/arcade/blob/5b66439eb4e396f105548150acffbaf062d9db76/src/Microsoft.DotNet.CoreFxTesting/build/core/test/Test.targets#L23

to: https://github.com/dotnet/arcade/blob/5b66439/src/Microsoft.DotNet.Arcade.Sdk/tools/Tests.props?

### PR DESCRIPTION
**Pull request by ViktorHofer**
Closes https://github.com/dotnet/arcade/issues/4105

With https://github.com/dotnet/arcade/pull/5314, framework and platform
traits don't need to be passed in manually anymore. Also removing the
obsolete vstest.runner.json as it isn't needed for VSTest.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
LGTM assuming it all works :)

**Review by akoeplinger — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Failures are because of https://github.com/dotnet/arcade/pull/5322.

File: eng/testing/xunit/xunit.targets [hunk 1]
```diff
@@ -1,11 +1,5 @@
 <Project>
   <PropertyGroup>
-    <TargetOSCategory Condition="'$(TargetOS)' == 'Windows_NT'">nonwindowstests</TargetOSCategory>
-    <TargetOSCategory Condition="'$(TargetOS)' == 'Linux'">nonlinuxtests</TargetOSCategory>
-    <TargetOSCategory Condition="'$(TargetOS)' == 'OSX'">nonosxtests</TargetOSCategory>
-    <TargetOSCategory Condition="'$(TargetOS)' == 'FreeBSD'">nonfreebsdtests</TargetOSCategory>
-    <TargetOSCategory Condition="'$(TargetOS)' == 'NetBSD'">nonnetbsdtests</TargetOSCategory>
```
Comment (am11) [inline eng/testing/xunit/xunit.targets:7]: Could these be removed from https://github.com/dotnet/runtime/blob/952eecabbdd57540f4fe72c490c794e79c3d0cf3/src/installer/test/Directory.Build.props?
(that will close https://github.com/dotnet/arcade/issues/4105 🙂)

Comment (ViktorHofer) [reply eng/testing/xunit/xunit.targets:7]: Unfortunately not. In installer we don't use XUnitExtensions afaik.

Comment (am11) [reply eng/testing/xunit/xunit.targets:7]: `XUnitExtensions` are used in installers: https://github.com/dotnet/runtime/blob/7abe42dc1123722ed385218268bb9fe04556e3d3/src/installer/test/TestUtils/TestUtils.csproj#L17 e.g.`PlatformSpecificAttribute` is used here: https://github.com/dotnet/runtime/blob/7abe42dc1123722ed385218268bb9fe04556e3d3/src/installer/test/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs#L170

Comment (ViktorHofer) [reply eng/testing/xunit/xunit.targets:7]: Yeah just noticed that :), thanks: https://github.com/ViktorHofer/runtime/commit/27048a24ecf74132f1114dd82b8f9e0171fee71c

**Pull request merged**